src/app/mat-card/mat-card.component.ts

```diff
@@ -8,17 +11,55 @@ import { SampleserviceService } from '../sampleservice.service';
   templateUrl: './mat-card.component.html',
   styleUrls: ['./mat-card.component.css']
 })
-export class MatCardComponent implements OnInit{
- product: Observable<cards[]> = of([])
+export class MatCardComponent implements OnInit {
+  cards: any
+
+
+
+
+  constructor(private serv: SampleserviceService, private router: Router) {
+
+  }
+
 
-  constructor(private serv:SampleserviceService){}
 
-  
-  
   ngOnInit(): void {
-  this.product = this.serv.getProductDetails()
- 
+    // this.serv.getclothing().pipe(takeUntil(this.destroyedData$)).subscribe((data)=>{
+    //   this.clothing=data
+
+    // })
+
+   
+    this.router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe((path: any) => {
+     
+      if (path.url.includes('clothing')) {
+
+        this.cards = this.serv.getclothing()
+
+      }
+
+      else if (path.url.includes('homeAppliances')) {
+        this.cards = this.serv.gethomeAppliance()
+
+      }
+      else if (path.url.includes('electronicAppliances')) {
+        this.cards = this.serv.getelectronicAppliance()
+
+      }
+      else {
+        this.router.navigate(['/goshopping'])
```

Comment:
why is navigation happening here, you have to just assign value to this.cards right..?